sample-apps/discount-functions-sample-app/web/frontend/vite.config.js

```diff
@@ -21,36 +20,55 @@ const proxyOptions = {
 };
 
 const host = process.env.HOST
-  ? process.env.HOST.replace(/https:\/\//, "")
-  : "localhost";
+  ? process.env.HOST.replace(/https:\/\//, '')
+  : 'localhost';
 
 // HMR doesn't work on Firefox using localhost, so you can temporarily get that to work by setting the
 // SHOPIFY_VITE_HMR_USE_POLLING env var when running this
 let hmrConfig;
 
-if (host === "localhost") {
+if (host === 'localhost') {
   hmrConfig = {
-    protocol: "ws",
-    host: "localhost",
+    protocol: 'ws',
+    host: 'localhost',
     port: 64999,
     clientPort: 64999,
   };
 } else {
   hmrConfig = {
-    protocol: "wss",
+    protocol: 'wss',
     host: host,
     port: process.env.FRONTEND_PORT,
     clientPort: 443,
   };
 }
+
 const root = new URL('.', import.meta.url).pathname;
 
+const ENV_KEYS = [
+  'SHOPIFY_API_KEY',
+  'SHOPIFY_ORDER_DISCOUNT_ID',
+  'SHOPIFY_SHIPPING_DISCOUNT_ID',
+  'SHOPIFY_PRODUCT_DISCOUNT_ID',
+];
+
+// Function IDs are populated in ../../.env by the Shopify CLI after being deployed
+if (process.env.npm_lifecycle_event === 'dev') {
```

Comment:
Do we only want to load the `.env` file in `dev`? Depending on how the developer runs their app, they might have a `.env` file in production rather than setting environment variables.

Reply:
Ah I didn't realize it was like this before! I don't think we should change it then. We can't make assumptions on CI/production since this is a starting template for a third-party developer and we don't know how they'll use it. I think it's best to stick to normal Vite things.